src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -57,31 +59,28 @@ public void emit(Event event)
       URI uri = uriExtractor.apply(event);
       HttpPostEmitter emitter = emitters.get(uri);
       if (emitter == null) {
-        synchronized (emitters) {
-          emitter = emitters.get(uri); // Double check that nobody else created emitter while we were waiting on lock
-          if (emitter == null) {
-            emitter = new HttpPostEmitter(
-                config.buildHttpEmitterConfig(uri.toString()),
-                client,
-                jsonMapper
+        emitter = emitters.computeIfAbsent(uri, u -> {
```

Comment:
As in general for ConcurrentMap the function passed to computeIfAbsent might be called several times and only ConcurrentHashMap provides guarantee that it will be executed at most once I suggest mentioning it in comment here.